client/state/themes/selectors.js

```diff
@@ -400,10 +405,12 @@ export function getThemeCustomizeUrl( state, theme, siteId ) {
 	}
 
 	if ( isJetpackSite( state, siteId ) ) {
+		// wpcom themes on jetpack sites use suffixed id
+		const themeIdSuffix = ( theme && theme.screenshots ) ? '-wpcom' : '';
```

Comment:
Same as https://github.com/Automattic/wp-calypso/pull/10526/files#r97169598 -- whatever we do to check if it's a `wpcom` theme, that check shouldn't be implicit and hidden but use a unified helper (util or selector).